src/legacy/ui/public/index_patterns/_index_pattern.js

```diff
@@ -260,6 +260,11 @@ export class IndexPattern {
     return this.fields.byName[this.timeFieldName];
   }
 
+  getFieldByName(name) {
+    if (!this.fields || !this.fields.byName) return;
+    return this.fields.byName[name];
+  }
+
```

Comment:
Added the new function to satisfy typescript, also a think it's a better way to access field settings